controllers/auth_config_controller.go

```diff
@@ -589,6 +590,84 @@ func (r *AuthConfigReconciler) translateAuthConfig(ctx context.Context, authConf
 	return evaluatorConfigByHost, nil
 }
 
+func (r *AuthConfigReconciler) addToCache(ctx context.Context, resourceNamespace, resourceId string, evaluatorConfigByHost map[string]evaluators.AuthConfig) (linkedHosts, looseHosts []string, err error) {
+	logger := log.FromContext(ctx)
+	linkedHosts = []string{}
+	looseHosts = []string{}
+
+	for host, evaluatorConfig := range evaluatorConfigByHost {
+		// check for host collision with another namespace
+		if cachedKey, found := r.Cache.FindId(host); found {
+			if cachedKeyParts := strings.Split(cachedKey, string(types.Separator)); cachedKeyParts[0] != resourceNamespace {
+				looseHosts = append(looseHosts, host)
+				logger.Info("host already taken in another namespace", "host", host)
+				continue
+			}
+		}
+
+		// add to cache
+		if err = r.Cache.Set(resourceId, host, evaluatorConfig, true); err != nil {
+			return
+		}
+
+		linkedHosts = append(linkedHosts, host)
+	}
+
+	return
+}
+
+func (r *AuthConfigReconciler) bootstrapCacheIndex(ctx context.Context) error {
+	r.cacheRebuild.Lock()
+	defer r.cacheRebuild.Unlock()
+
+	if !r.Cache.Empty() {
+		return nil
+	}
+
+	logger := r.Logger.WithName("bootstrap")
+	logger.Info("building cache index")
+
+	authConfigList := api.AuthConfigList{}
+	if err := r.List(ctx, &authConfigList); err != nil {
+		return err
+	}
+
+	sort.Sort(authConfigList.Items)
+
+	ctx = log.IntoContext(ctx, logger)
+	denyAll := evaluators.AuthConfig{
+		AuthorizationConfigs: []auth.AuthConfigEvaluator{evaluators.NewDenyAllAuthorization(ctx, "deny-all", "")},
+		DenyWith:             evaluators.DenyWith{Unauthorized: &evaluators.DenyWithValues{Code: 503, Message: &json.JSONValue{Static: "Busy"}}},
+	}
+
+	for _, authConfig := range authConfigList.Items {
+		if len(authConfig.Status.Summary.HostsReady) == 0 { // unfortunately we cannot use arbitrary field selectors for custom resources yet - https://github.com/kubernetes/kubernetes/issues/51046
+			continue
+		}
+
+		evaluatorConfigByHost := make(map[string]evaluators.AuthConfig)
+		for _, host := range authConfig.Spec.Hosts {
+			evaluatorConfigByHost[host] = denyAll
+		}
+
+		authConfigName := types.NamespacedName{Namespace: authConfig.Namespace, Name: authConfig.Name}
+		logger.V(1).Info("building cache index", "authconfig", authConfigName)
+
+		_, _, err := r.addToCache(
+			log.IntoContext(ctx, logger.WithValues("authconfig", authConfigName)),
+			authConfig.Namespace,
+			authConfigName.String(),
+			evaluatorConfigByHost,
```

Comment:
I think I need of context to understand the crafted before "deny all" policy added to the cache here

Reply:
Ok, just read it... then it willl update with the true values from the `AuthConfig`s replacing the bootstrapped deny all one :+1: